src/cluster_info.rs

```diff
@@ -174,16 +174,16 @@ impl ClusterInfo {
         let id = node_info.id;
         me.gossip.set_self(id);
         me.insert_info(node_info);
-        me.push_self();
+        me.push_self(None);
         me
     }
-    pub fn push_self(&mut self) {
+    pub fn push_self(&mut self, bank: Option<&Arc<Bank>>) {
         let mut my_data = self.my_data();
         let now = timestamp();
         my_data.wallclock = now;
         let mut entry = CrdsValue::ContactInfo(my_data);
         entry.sign(&self.keypair);
-        self.gossip.refresh_push_active_set();
+        self.gossip.refresh_push_active_set(bank);
```

Comment:
@garious @sagar-solana @carllin 

This should be overlapped with the data necessary for generating a leader schedule.  I think we should compute the active set once from the right bank and then either query that bank for it via a function, or pass it around instead of the bank (preferred), since it would be easier to test with.  `HashMap<Pubkey, u64>`, is all you would need I imagine.

Reply:
Check out the new ActiveStakers object that I just added to the repo earlier today!

Reply:
@garious, I think I'll have to stick with this implementation for now and maybe update it to use a BankForks and grab the "active" bank further down the line. ActiveStakers may not contain _all_ nodes that gossip should be _gossiping_ with.

Reply:
I see.  I was confused by the use of the term active set. Luckily, I just removed all instances of that from LeaderScheduler.  Gossip may have it. :)

Reply:
@sagar-solana if the nodes are not staked, they can have a minimum weight and treated as a separate class.  I don't think it makes a lot of sense to use an account balance for non-staked nodes.

Reply:
@aeyakovenko, yeah this is something I need to figure out now. I assumed account balance was sufficient but only just realized it's not their stake. What do I use to figure out stakes instead?

Reply:
@sagar-solana i can short some tokens and mess with the network :).  Staking requires a bigger investment.

Reply:
@aeyakovenko yup I think I understand that but I also just realized that I'm using account balances for both Avalanche and now for Gossip. I think we're using vote accounts as the staked tokens but I'm not sure if there's a simple way to get to them. 

Maybe I'll add something like Bank.get_stake(pubkey) that returns the vote account balance or 0? Also maybe I'll do that in a separate pr.

Reply:
@aeyakovenko looks like I can use what @garious did with stakes using bank.vote_states and use the account balances of those vote accounts instead of the node's bank balance.

Reply:
Sadly, it's not actually the balances of vote accounts.  It's the balances of the `staker_id` accounts that are held by vote accounts.

Reply:
@garious, that's confusing. If I'm looking at the code correctly right now staker_id and node_ids are basically the same all the time, so I'd end up with effectively the same thing as bank.get_balance(node_id) ?

Reply:
Do not get the balance of `node_id`. That's the ephemeral fullnode keypair.  Yes, things are in a very confusing state.  You should see what it looked like before!

Reply:
@garious, Yeah I'd pick the balance of the staker_id but I need to find which staker based on the node_id. It's more like things have been changing so quickly lately, what I understood at some point is now totally different :D.
Would you rather have me address all the outstanding comments (Option and Correct Stakes) before merging this or do that as a hotfix and cleanup to this? The Bank stakes thing will apply to avalanche too so I was hoping to do it as a second pass. Thoughts?

Reply:
Regarding things changing quickly, your best bet is to not try to build upon things that are clearly a mess. Those things will either be changed by you or under your feet. I recommend the former! Shave those yaks!

Second pass is fine. Just don't leave a mess for someone else (probably me) to clean up.